GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -2866,7 +2866,8 @@ private async Task PopulateToolbarAsync(SubmoduleInfoResult result, Cancellation
             // then refresh all items at once with a single switch to the main thread
             ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
-                var loadDetails = newItems.Select(e => e.loadDetails).Where(e => e is not null);
+                await TaskScheduler.Default;
+                var loadDetails = newItems.Select(e => e.loadDetails).WhereNotNull();
```

Comment:
The previous code would be better for all cases except ones where `loadFunc` is improperly written. Were there any such cases?

Reply:
loadFunc launches Git commands that will run on the UI thread when setting up this menu, so I believe this makes sense

Reply:
reverting, loadfunc is normally null